thundra/thundra_agent.py

```diff
@@ -123,4 +144,8 @@ def checkAndHandleWarmupRequest(self, event):
                     return True
             return False
 
+    def timeout_handler(self, signum, frame):
```

Comment:
"signum" and "frame" args are unused here.
it might be ok since signal function expects its handler to be having those arguments. However, in that case it will be wise to check signum againist: signal.SIGALRM to support explicit usage. 

 Also, it smells a little bit code duplication because of the code above in the except:
`self.data['error'] = e`
                ` self.execute_hook('after:invocation', self.data)`
                 `self.reporter.send_report()`